larq_compute_engine/tflite/tests/bconv2d_test.cc

```diff
@@ -351,13 +368,64 @@ TEST_P(BConv2DOpTest, SimpleTest) {
     return float_list[index];
   };
 
-  std::generate(std::begin(input_data), std::end(input_data), rand_generator);
+  std::generate(std::begin(bconv_input_data), std::end(bconv_input_data),
+                rand_generator);
   std::generate(std::begin(filters_data), std::end(filters_data),
                 rand_generator);
   std::generate(std::begin(channel_multipliers), std::end(channel_multipliers),
                 float_generator);
   std::generate(std::begin(bias_data), std::end(bias_data), float_generator);
 
+  int tflite_input_height = input_height;
```

Comment:
this should be the exact calculation that Larq does to pad the QuantConv with ones and in Larq i believe the TF padding op is used so this also needs to use the same TF ops and the same padding layout (otherwise how Larq is used the train the weights will be different than how the inference engine is implemented) so instead of manually calculating the padding sizes which might differ from how Larq does it, we need to use the TF ops that Larq uses here. I have implemented example of using C++ API of TF ops in this test.

Reply:
I agree that its better to use TF ops. However I could only find this `SingleOpModel` testing interface where you can not give it two ops in a row, thats why I did it manually.

Do you know of a way that we can build a multi-op model in this testing framework?

Reply:
in case that there is not multi-op model, you can pass the output of one op as an input to another single op model? but can you check first if there is really no multi-op model? how TF lite tests the fused ops?

Reply:
New commit uses two `SingleOpModel`s. We still have to manually compute the padding amounts since the `Pad` op wants explicit paddings.